### ISSUE TITLE
No longer Win7 compatible?

### ISSUE THREAD

**Issue by TruthInFiction** [MrAlex94/Waterfox#3609]
### What happened?

The website still lists Windows 7 as the minimum, but when trying to auto-update from 6.0.5 to latest version it is reported that my operating system is unsupported.  I tried downloading the installer and updating that way but it reports that Windows 10 is required.

### Reproducible?

- [ ] I have checked that this issue cannot be reproduced on Mozilla Firefox.

### Version

G6

### What platform are you seeing the problem on?

Windows

### Relevant log output

```shell

```

**Comment by hawkeye116477** [comment]
That's right. Last FF for W7 is 115, so Waterfox 6.0.20 is also last version for W7. 6.5.0 is based on newer Firefox, so W10 is required.

**Comment by TruthInFiction** [comment]
That's disappointing.  I suppose that means I'll need to look for another browser for my older machine.  They ought to update the website to reflect the new minimum requirements.

**Comment by hawkeye116477** [comment]
I think that you should rather prepare to switch to some Linux (maybe Ubuntu or Zorin) if you can't upgrade to W10  :smile:.

**Comment by Ibuprophen** [comment]
Same incompatibility for Windows 8.1... :(

~Ibuprophen

**Comment by TruthInFiction** [comment]
@hawkeye116477 The main reason I was hoping for Win 7 compatibility to continue is for my retro pc.  I've got a system with Linux Mint and Windows 10, although I've mostly abandoned Windows 10 at this point with no plans of downgrading to Windows 11.

**Comment by IS4Code** [comment]
I hope this will be fixed eventually. [This page](https://www.waterfox.net/download/) lists Windows 7 as minimum OS, and trying to update redirects me to a [non-existent page](https://www.waterfox.net/support/) (and [same](https://www.waterfox.net/support/desktop/system-requirements/) for the installer). This makes me assume this is unintended.

**Comment by IdrisQe** [comment]
The release notes for 6.5.0 mention "A re-jig of the system requirements to run Waterfox. We now expect the same minimum as Windows 10"
I feel like I can safely assume that means they also expect Windows 10 at minimum. Sorry to say this probably isn't a bug.

That said, if you're still running Windows 7 or 8.1, you're likely comfortable using an OS that has had no security updates in almost 5 or 2 years respectively, so you're probably fine to stick to the previous Waterfox version until the next big wave of internet overhauls eventually break it (though in my opinion, if you must browse the internet at all on an OS that old, you should probably be using a much more hardened browser to begin with.)

That said, I would be happy to be proven wrong here and learn this _is_ a bug, because I can respect still using older Windows versions. I'm still not totally happy with Windows 10, and my new Laptop will come with 11, which is... _Less than ideal._ Personally hoping I can get more comfortable with some distribution of Linux as a daily driver, because if 10 was a step too far, 11 is running straight off the edge of a cliff Looney Tunes style.

**Comment by asciideer** [comment]
as IS4Code mentioned, the download page still lists windows 7 as the minimum required version, which is incorrect; *that* should be fixed if there are no plans to support it.

**Comment by Squall-Leonhart** [comment]
does it install an work with vxkex?

**Comment by ghost** [comment]
> That's right. Last FF for W7 is 115, so Waterfox 6.0.20 is also last version for W7. 6.5.0 is based on newer Firefox, so W10 is required.

This portable version 6.5.0 seems to work. But seems to be modded.
https://github.com/adeii/supermium-portable/releases/download/F130/Waterfox-portable-6.5.0-x64.7z

**Comment by BenFenner** [comment]
I ran into this issue today. The Minimum System Requirements on [the download page](https://www.waterfox.net/download) show Windows 7, leading someone else to suggest Waterfox for a sidekick browser on 4 of my 5 machines. Alas, the minimum system requirements are outdated as of 6.5.0. Fixing the web site seems to be the expected solution.

And that [404 page](https://www.waterfox.net/support) should be populated, or links to it removed.

**Comment by Bushroott** [comment]
> That said, if you're still running Windows 7 or 8.1, you're likely comfortable using an OS that has had no security updates in almost 5 or 2 years respectively

The latest Windows 7 update to date was released on January 14, 2025.
Firefox/Waterfox does not support Windows 7 solely for "political" reasons. There is nothing critically needed for the browsers in Windows 10 that is not in Windows 7.

Drop the Foxes, use PaleMoon.

**Comment by IdrisQe** [comment]
> > That said, if you're still running Windows 7 or 8.1, you're likely comfortable using an OS that has had no security updates in almost 5 or 2 years respectively
> 
> The latest Windows 7 update to date was released on January 14, 2025. Firefox/Waterfox does not support Windows 7 solely for "political" reasons. There is nothing critically needed for the browsers in Windows 10 that is not in Windows 7.
> 
> Drop the Foxes, use PaleMoon.

Are you trolling? All editions of Windows 7 stopped recieving _even extended support_ over 5 years ago. If you're getting new updates, they arent official.

**Comment by Bushroott** [comment]
> Are you trolling? All editions of Windows 7 stopped recieving _even extended support_ over 5 years ago. If you're getting new updates, they arent official.

The updates are completely official, from Microsoft. They are intended for Windows Server 2008 R2, but are perfect for Windows 7 - internally these systems are totally identical.
To install updates, you will need an ESU license.
Moreover, Vista can be updated in the same way.

**Comment by IdrisQe** [comment]
> > Are you trolling? All editions of Windows 7 stopped recieving _even extended support_ over 5 years ago. If you're getting new updates, they arent official.
> 
> The updates are completely official, from Microsoft. They are intended for Windows Server 2008 R2, but are perfect for Windows 7 - internally these systems are totally identical. To install updates, you will need an ESU license. Moreover, Vista can be updated in the same way.

Yeah sorry, I'm not counting using updates from a seperate OS as Windows 7 "still recieving updates", even if they're almost identical internally. That's just a hacky workaround to keep using a long-outdated OS, and is by no means official support for Windows 7 itself.

**Comment by Bushroott** [comment]
>That's just a hacky workaround to keep using a long-outdated OS, and is by no means official support for Windows 7 itself.

There is nothing hackish about this way.
Standalone update installers are simply downloaded from the Microsoft Download Center and installed.
Important detail: the installers do not indicate anywhere that they are intended specifically for Windows Server. It is simply written - Windows.

**Comment by Squall-Leonhart** [comment]
whether you can update windows 7 or not has nothing to do with this bug.

**Comment by TruthInFiction** [comment]
It's crazy how some people just take offense at the fact that some of us want to continue using software we like instead of embracing the ever more exploitative slop Microsoft is pushing out.  

I mean, I switched to Linux Mint for my main system well over a year ago, but I just liked the look and feel of Windows 7, and it's somehow a crime against humanity to want to continue using it.  It's a shame that Waterfox has dropped support, but they should update all instances where Windows 7 is listed as the minimum requirement.  The fact that another issue was raised about this just recently ought to be an indicator of this.

**Comment by Bushroott** [comment]
> I mean, I switched to Linux Mint for my main system well over a year ago, but I just liked the look and feel of Windows 7, and it's somehow a crime against humanity to want to continue using it.

Linux is built on 60-year-old standards, for 60-year-old computers that were kept in secure rooms and run by scientists. This operating system is not as safe in the hands of the average user as a howitzer or a bazooka.
Giving a howitzer to civilians is a crime.
Using Windows 7 is not a crime.

**Comment by IdrisQe** [comment]
> It's crazy how some people just take offense at the fact that some of us want to continue using software we like instead of embracing the ever more exploitative slop Microsoft is pushing out.
> 
> I mean, I switched to Linux Mint for my main system well over a year ago, but I just liked the look and feel of Windows 7, and it's somehow a crime against humanity to want to continue using it. It's a shame that Waterfox has dropped support, but they should update all instances where Windows 7 is listed as the minimum requirement. The fact that another issue was raised about this just recently ought to be an indicator of this.

Yeah, the documentation needs to be updated everywhere for certain... But I understand it not being a priority given Windows 7 doesn't exactly have a booming installbase these days.

Trust me, I would be using Windows 7 still too if I thought it was a legitimate option. But I'm not willing to spoof my system into getting updates from a server version in order to have a secure(ish) system, and if I was to use it I'd be giving my performance at a big detriment due to the lack of support for things like DirectX12 and increasingly poor driver support from newer hardware.

It's not a crime to find workarounds for all this, but requesting that Waterfox devs put in extra time to ensure that an OS that has not been officially updated in over 5 years is supported by new versions of their browser, is a bit ridiculous. If you're okay using an old OS with all the problems that entails, you should be okay using an old browser too. Or find a fork of Firefox that does have Win7 support to this day (I'm sure there's at least 1 or 2 given how many forks there are)

I get it though, as someone who has SEVERE issues getting used to changes (for reference, I still have my Waterfox set up to have a UI as close as I can to Firefox's v4 UI, and have done many ill-advised tweaks to my Windows 10 to make it function as similarly to 7 as possible):
I clung to Windows 7 as long as I could. I even clung to 8.1 for longer than I should have. I only updated when getting new computers that came with a newer OS and was too cheap to buy a license for an old version. And now I'm clinging to 10 desperately, despite knowing in just a few months it'll be losing support too. When that happens, I'll either migrate to 11 and spend weeks tweaking it until I consider it usable (probably still hating it with every fiber of my being as has been the case every time I've had to use a computer that's infected with it) or I'll finally switch to Linux, which I tried once before, but couldn't get used to the differences.

Anyway, this conversation has long since derailed.

Again, I'd reccommend either looking for other Firefox forks that still have Win7 support, or just use old versions for now. Given how many people still use Windows 7, I wouldn't be surprised if someone was already working on a modern Windows 7 fork of Waterfox itself. Everyone has to understand that as Operating Systems move forward, not every version can be supported by every piece of software forever. There are too many architectural differences, it would lead to a more and more bloated codebase.
As someone who has had the misfortune of programming since a young age, I _wish_ it was easy to keep a codebase working with multiple versions of a dependant platform, but it just isn't (well, except in very specific circumstances, but this isn't one of those).

**Comment by ioc2e3** [comment]
> ... if 10 was a step too far, 11 is running straight off the edge of a cliff Looney Tunes style.

Totally agree ...

.

**Comment by ioc2e3** [comment]
> ... . Using Windows 7 is not a crime.

Totally agree ...

.

**Comment by scorpion451** [comment]
So, any progress on getting a build for windows 7, or are we considered non-entities for not wanting to pay for the privilege of perpetually beta testing a buggy mess of an operating system that microsoft insists is an upgrade?

**Comment by MrAlex94** [comment]
Hi everyone,

I completely understand the disappointment around dropping support for older operating systems (especially a popular one like Windows 7). I know how frustrating it can be when software you rely on is no longer compatible with your system, especially when that system is still serving you well.

This wasn't a decision I made lightly. Even if I personally wanted to maintain support for these older OS versions, the reality is that I simply don't have the development capacity to properly support and test across all these platforms while also moving the project forward with new features and security updates.

Supporting older operating systems requires significant ongoing effort - not just initial compatibility work, but continuous testing, debugging platform-specific issues, and ensuring security patches work correctly across all supported versions. With limited time and resources, I have to make difficult choices about where to focus development efforts.

I know there are some devs doing the work for it - but I'm already spread so thin as is it's not really something I can promise or give for Waterfox.

**Comment by BenFenner** [comment]
Closing this ticket completely misses the point. As described in the initial report, and reiterated in my initial post here: https://github.com/BrowserWorks/waterfox/issues/3609#issuecomment-2578944019

**The issue is not about drumming up support for Windows 7.**

The issue is that the download page of the web site shows Windows 7 as a compatible OS, when it is not. It can't take that much dev resources to fix the verbiage on the download page, can it?

<img width="1358" height="1358" alt="Image" src="https://github.com/user-attachments/assets/f1c4df91-400c-45dc-8394-86c3062d8319" />

**Comment by lixkote** [comment]
> I think that you should rather prepare to switch to some Linux (maybe Ubuntu or Zorin) if you can't upgrade to W10 😄.

ubuntu is a total unstable pile of shit, zorin is sketchy. better go with mint cinnamon instead

**Comment by lixkote** [comment]
> > > That said, if you're still running Windows 7 or 8.1, you're likely comfortable using an OS that has had no security updates in almost 5 or 2 years respectively
> > 
> > 
> > The latest Windows 7 update to date was released on January 14, 2025. Firefox/Waterfox does not support Windows 7 solely for "political" reasons. There is nothing critically needed for the browsers in Windows 10 that is not in Windows 7.
> > Drop the Foxes, use PaleMoon.
> 
> Are you trolling? All editions of Windows 7 stopped recieving _even extended support_ over 5 years ago. If you're getting new updates, they arent official.

If you haven't figured that out yet, some people are power users, and simply don't give 2 shits about m$ updates, they know how to keep their pcs clean and they don't do any mission-critical stuff on their computers.

**Comment by lixkote** [comment]
> trolling

take this then 

<img width="1111" height="709" alt="Image" src="https://github.com/user-attachments/assets/23defb2d-553d-4abd-af59-0511adf466fb" />

**Comment by IdrisQe** [comment]
_Oh my god people are still complaining about this?_

Okay yeah, I get it, some people are still going to stay on Windows 7 and get unofficial updates or whatever.
Still doesn't mean anyone is required to support a 16 year old operating system which has less than 3% market share among non-business users last I checked.

I'm going to finally unfollow this trainwreck of an issue now, just so I can stop recieving notifications from obsessed people.

**Comment by eglacias** [comment]
Edit: the page says VxKex works with Firefox :-)


> does it install an work with vxkex?

exactly this: haven't tried yet but most things run fine with vxKex extensions. gonna test it one day...

**Comment by lixkote** [comment]
Just use SecondSystem or vxkex tbh, on w8.1 secondsystem basically makes 90% of w10 apps just run

and for the start menu, you can use https://github.com/lixkote/DirectStart

**Comment by eglacias** [comment]
Or I still use classic start menu, which is now open shell, so I can actually see immediately all the programs I’ve installed again, lol! Works great on windows 10 too. I have a dual boot system for the twice in a year times I need to use windows 10 for something.

SecondSystem looks like it was designed specifically for windows vista and maybe more outdated? than 

https://github.com/i486/VxKex
VxKex which Now has a decently maintained version it seems like it’s a more complete solution.
It is confirmed to work with Firefox! That means it will work with Waterfox

**Comment by adonios77** [comment]
Personally, I prefer to eat yesterday's food from the fridge, instead of fresh hot shit.
 Bon appetit.  ;)
